### PR TITLE
Node Fixes

### PR DESCRIPTION
**Pull request by jrhea**
fixing issues where:
- node.PeerId was wrong.
- pubsubPeerDiscovery intercepting bunnymen protocol messages and throwing errors

added custom webrtc-star endpoint

### CODE REVIEW

File: packages/bunnymen-core/src/dataset.ts [hunk 1]
```diff
@@ -135,7 +135,7 @@ export class Dataset<TData = any, TNewData = TData>
   async init() {
     // download payloads after a peer sends a new CID
     this.node.on('receivedMessage', async (receivedCID) => {
-      if (receivedCID === this.currentCID) {
+      if (receivedCID === this.currentCID || receivedCID.length !== 46) {
```
Comment (ryangoree) [inline packages/bunnymen-core/src/dataset.ts:138]: nice catch

File: packages/bunnymen-core/src/node.ts [hunk 2]
```diff
@@ -240,17 +240,21 @@ export class Node extends EventEmitter {
...
     return createLibp2p({
+      peerId: this._opts.peerId,
       addresses: {
         // Add the signaling server address, along with our PeerId to our multiaddrs list
         // libp2p will automatically attempt to dial to the signaling server so that it can
         // receive inbound connections from other peers
         listen: [
           //'/dns4/wrtc-star1.par.dwebops.pub/tcp/443/wss/p2p-webrtc-star',
           //'/dns4/wrtc-star2.sjc.dwebops.pub/tcp/443/wss/p2p-webrtc-star',
+          '/dns4/bunnymen.delvelabs.xyz/tcp/443/wss/p2p-webrtc-star/',
```
Comment (ryangoree) [inline packages/bunnymen-core/src/node.ts:251]: 🔥

**Review by ryangoree — APPROVED** [review]
nice

File: packages/bunnymen-core/src/dataset.ts (hunk 1 above)
Comment (jrhea) [reply packages/bunnymen-core/src/dataset.ts:138]: ya that was a fun one

File: packages/bunnymen-core/src/node.ts (hunk 2 above)
Comment (jrhea) [reply packages/bunnymen-core/src/node.ts:251]: oh ya. i forgot that i added this too. i will add it to the PR summary

**Pull request merged**